iiod/ops.c

```diff
@@ -760,8 +773,13 @@ static void rw_thd(struct thread_pool *pool, void *d)
 			for (thd = SLIST_FIRST(&entry->thdlist_head);
 					thd; thd = next_thd) {
 				next_thd = SLIST_NEXT(thd, dev_list_entry);
-				if (thd->active && thd->is_writer &&
-						thd->nb < sample_size)
+
+				if (!thd->active || !thd->is_writer)
+					continue;
+
+				if (ret < 0)
```

Comment:
I think I agree with codacy, this check will always be the same, since there is a check above that breaks out of the while loop...

Reply:
Yeah, I guess you need to remove the outer check...

Reply:
Yes that was supposed to be removed, good catch.